src/anomalib/models/video/fuvas/torch_model.py

```diff
@@ -0,0 +1,218 @@
+
+        Returns:
+            Union[torch.Tensor, Tuple[torch.Tensor, torch.Size]]: Features during
+                training, or tuple of (features, feature_shapes) during inference.
+        """
+        with torch.no_grad():
+            self.feature_extractor.eval()
+            batch = torch.permute(batch, (0, 2, 1, 3, 4))
+            out_dict = self.feature_extractor(batch)
+            if self.layer.startswith("features"):
+                out = torch.permute(out_dict[self.layer], (0, 1, 4, 2, 3))
+            else:
+                out = out_dict[self.layer]
+            # pool
+            if self.spatial_pool:
+                if len(out.shape) == 5:
+                    pool_features = F.avg_pool3d(out, (1, self.pooling_kernel_size, self.pooling_kernel_size))
+                elif len(out.shape) == 4:
+                    pool_features = F.avg_pool2d(out, (self.pooling_kernel_size, self.pooling_kernel_size))
+                else:
+                    pool_features = F.avg_pool1d(out, self.pooling_kernel_size)
+                feature_shape = pool_features.shape
+            else:
+                feature_shape = list(out.shape)
+                fea_vector = out.reshape(feature_shape[0], -1)
+                pool_features = F.avg_pool1d(fea_vector, self.pooling_kernel_size)
+                feature_shape[1] = feature_shape[1] // self.pooling_kernel_size
+
+        features = pool_features.cpu().reshape(feature_shape[0], -1)
+        return features, feature_shape
+
+    def forward(self, batch: torch.Tensor) -> torch.Tensor | InferenceBatch:
+        """Compute anomaly predictions from input images.
+
+        Args:
+            batch (torch.Tensor): Input images with shape
+                ``(batch_size, clip_len,channels, height, width)``.
+
+        Returns:
+            Union[torch.Tensor, InferenceBatch]: Model predictions. During
+                training returns features tensor. During inference returns
+                ``InferenceBatch`` with prediction scores and anomaly maps.
+        """
+        feature_vector, feature_shapes = self.get_features(batch)
+        anomaly_map = None
+        if self.do_seg:
+            pred_score, anomaly_map = self.score(feature_vector, feature_shapes)
+        else:
+            pred_score = self.score(feature_vector, feature_shapes)
+        if anomaly_map is not None:
+            anomaly_map_stack = []
+            for map_one in anomaly_map:
+                map_one_np = map_one.numpy()
+                anomaly_map_stack.append(resize(map_one_np, tuple(batch.shape[-2:])))
+            anomaly_map_stack = np.stack(anomaly_map_stack)
+            anomaly_map_stack = torch.from_numpy(anomaly_map_stack)
+            anomaly_map_stack = torch.unsqueeze(anomaly_map_stack, 1).to(batch.device)
+        pred_score = pred_score.to(batch.device)
+        return InferenceBatch(pred_score=pred_score, anomaly_map=anomaly_map_stack)
```

Comment:
Here as well, would it be possible to remove numpy conversions, and implement every line in torch?